### ISSUE TITLE
Latest `kubebuilder-declarative-pattern` version stuck on endless reconcile loop for guestbook-operator

### ISSUE THREAD

**Issue by asilverman** [kubernetes-sigs/kubebuilder-declarative-pattern#176]
**What happened**:
Following the instructions of the latest version of the [walkthrough](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/tree/master/docs/addon/walkthrough#readme) results in a broken state, the operator enters an endless loop of create/delete of the resources defined in the `channels/packages/guestbook/0.1.0` folder. So the package is never in a `Ready` state.

**What you expected to happen**:
I expected that the package would enter a stable state where it is `Ready`

**How to reproduce it (as minimally and precisely as possible)**:

1) Clone https://github.com/asilverman/guestbook-operator
2) Create a vanilla kind cluster with k8s version 1.12.1 (latest)
From the `guestbook-operator` folder, run (in sequence):
3) `make generate`
4) `make install`
5) `kubectl apply -n kube-system -f config/samples/addons_v1alpha1_guestbook.yaml`
6) `make run`

Inspect the logs of the operator and notice how it enters an endless loop

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
clientVersion:
  buildDate: "2020-08-13T16:12:48Z"
  compiler: gc
  gitCommit: 9f2892aab98fe339f3bd70e3c470144299398ace
  gitTreeState: clean
  gitVersion: v1.18.8
  goVersion: go1.13.15
  major: "1"
  minor: "18"
  platform: linux/amd64
serverVersion:
  buildDate: "2021-05-21T23:01:33Z"
  compiler: gc
  gitCommit: 5e58841cce77d4bc13713ad2b91fa0d961e69192
  gitTreeState: clean
  gitVersion: v1.21.1
  goVersion: go1.16.4
  major: "1"
  minor: "21"
  platform: linux/amd64
```
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel (e.g. `uname -a`):
```
Linux OFFICE-DT 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
- Go version (e.g. `go version`): `go version go1.16.5 linux/amd64`

- Others:

**Comment by asilverman** [comment]
gentle ping @atoato88

**Comment by atoato88** [comment]
Thank you to mention me.
I think current walkthrough docs is a little old, because some updates were merged into master branch.
Now, #173 is open for fixing some errors.

And we can update the docs and code of guestbook-operator once that PR is merged.

**Comment by asilverman** [comment]
@atoato88  I agree, however I tried using the latest version from #173, however the issue persists. I think the endless create/delete loop is unrelated. Could you please take a look?

**Comment by atoato88** [comment]
@asilverman 
I've created a PR #179 which fixes a problem in this issue.
For fixing endless loop, we need to [reomve `declarative.WithPreserveNamespace()`](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/179/files#diff-8a9e8bde8983509c3857375508fa2140508d4524d3b09ee326ee9accfbf55de9L150) from `Reconciler.Init` I think.
Please try that.

**Comment by asilverman** [comment]
@atoato88 I worry that the problem has not been fixed, notice how a describe on the frontend deployment is garbage collected:
```
Warning  OwnerRefInvalidNamespace  1s    garbage-collector-controller  ownerRef [addons.example.org/v1alpha1/Guestbook, namespace: guestbook, name: guestbook-sample, uid: d1929039-2182-4ab6-a863-afb0f890ee34] does not exist in namespace "guestbook"
```
